src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -4782,6 +4782,8 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
         source = os_disk_image = data_disk_images = None
         if managed_image is not None:
             source = GalleryArtifactVersionFullSource(id=managed_image)
+        if virtual_machine is not None:
+            source = GalleryArtifactVersionFullSource(virtual_machine_id=virtual_machine)
```

Comment:
Accord to the `L2425-L2434` in file `vm/_validators.py`, the `virtual_machine` value is also assigned to `managed_image`. Consequently, the line `source = GalleryArtifactVersionFullSource(id=managed_image)` will be executed even when a `--virtual-machine` argument is provided. This overlap could introduce potential risks.